Foundation/FileManager+Win32.swift

```diff
@@ -689,27 +690,52 @@ extension FileManager {
                 } while FindNextFileW(h, &ffd)
                 return files
             }
-            while let url = _stack.popLast() {
-                if url.hasDirectoryPath && !_options.contains(.skipsSubdirectoryDescendants) {
-                    guard let dirContents = contentsOfDir(directory: url)?.reversed() else {
-                        if let handler = _errorHandler {
-                           let dirFSR = url.withUnsafeFileSystemRepresentation { $0.flatMap { fsr in String(utf8String: fsr) } }
-                           let keepGoing = handler(URL(fileURLWithPath: dirFSR ?? ""),
-                               _NSErrorWithWindowsError(GetLastError(), reading: true))
-                           if !keepGoing { return nil }
-                        }
-                        continue
-                    }
-                    _stack.append(contentsOf: dirContents)
+
+            func findFirstValidFile() -> URL? {
+              while let url = _stack.popLast() {
+                let dirFSR = url.withUnsafeFileSystemRepresentation { $0.flatMap { fsr in String(utf8String: fsr) } } ?? ""
+                if !FileManager.default.fileExists(atPath: dirFSR, isDirectory: nil) {
```

Comment:
Huh?  Should `dirFSR` be renamed?  If `isDirectory` is `nil`, then we do not know if it is a file or directory, so it seems to be misleading.